### PR TITLE
Check for v1 IngressClass resource

### PR DESCRIPTION
**Pull request by swetharepakula**
Check for v1 IngressClass resource instead of v1beta1.

### CODE REVIEW

File: cmd/glbc/app/init.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ func IngressClassEnabled(client kubernetes.Interface) bool {
 	klog.V(2).Info("Checking if Ingress Class API exists")
 
 	err := wait.Poll(3*time.Second, 5*time.Minute, func() (bool, error) {
-		resourceList, err := client.Discovery().ServerResourcesForGroupVersion("networking.k8s.io/v1beta1")
+		resourceList, err := client.Discovery().ServerResourcesForGroupVersion("networking.k8s.io/v1")
```
Comment (liggitt) [inline cmd/glbc/app/init.go:69]: shouldn't there be a test that would have caught this?

Comment (swetharepakula) [reply cmd/glbc/app/init.go:69]: We depend on our e2e tests for validating this logic

Comment (liggitt) [reply cmd/glbc/app/init.go:69]: but... they didn't catch the issue before, so is an update to those needed?

Comment (swetharepakula) [reply cmd/glbc/app/init.go:69]: This logic is flag gated, and the flag is not enabled in the OSS tests which is why this was not caught.

Comment (liggitt) [reply cmd/glbc/app/init.go:69]: ok, thanks, will look for tests in the environments where the flag is enabled

**Review by freehan — COMMENTED** [review]
/lgtm
/approve

**Comment by swetharepakula** [comment]
/hold cancel
Canceling hold with the additional review.

**Pull request merged**